pkg/s3select/input.go

```diff
@@ -116,95 +153,34 @@ type Options struct {
-	myReader := opts.ReadFrom
-	var tempBytesScanned int64
-	tempBytesScanned = 0
-	switch opts.Compressed {
-	case "GZIP":
-		tempBytesScanned = opts.StreamSize
-		var err error
-		if myReader, err = gzip.NewReader(opts.ReadFrom); err != nil {
-			return nil, ErrTruncatedInput
-		}
-	case "BZIP2":
-		tempBytesScanned = opts.StreamSize
-		myReader = bzip2.NewReader(opts.ReadFrom)
-	}
+// JSONOptions options are passed to the underlying encoding/json reader.
+type JSONOptions struct {
 
-	// DelimitedReader treats custom record delimiter like `\r\n`,`\r`,`ab` etc and replaces it with `\n`.
-	normalizedReader := ioutil.NewDelimitedReader(myReader, []rune(opts.RecordDelimiter))
-	progress := &statInfo{
-		BytesScanned:   tempBytesScanned,
-		BytesProcessed: 0,
-		BytesReturned:  0,
-	}
-
-	reader := &Input{
-		options: opts,
-		reader:  csv.NewReader(normalizedReader),
-		stats:   progress,
-	}
-	reader.firstRow = nil
-
-	reader.reader.FieldsPerRecord = -1
-	if reader.options.FieldDelimiter != "" {
-		reader.reader.Comma = rune(reader.options.FieldDelimiter[0])
-	}
-
-	if reader.options.Comments != "" {
-		reader.reader.Comment = rune(reader.options.Comments[0])
-	}
+	// Name of the table that is used for querying
+	Name string
 
-	// QuoteCharacter - " (defaulted currently)
-	reader.reader.LazyQuotes = true
+	// ReadFrom is where the data will be read from.
+	//ReadFrom io.Reader
+	ReadFrom io.Reader
 
-	if err := reader.readHeader(); err != nil {
-		return nil, err
-	}
+	// If true then we need to add gzip or bzip reader.
+	// to extract the csv.
+	Compressed string
 
-	return reader, nil
-}
+	// SQL expression meant to be evaluated.
+	Expression string
 
-// ReadRecord reads a single record from the . Always returns successfully.
-// If the record is empty, an empty []string is returned.
-// Record expand to match the current row size, adding blank fields as needed.
-// Records never return less then the number of fields in the first row.
-// Returns nil on EOF
-// In the event of a parse error due to an invalid record, it is logged, and
-// an empty []string is returned with the number of fields in the first row,
-// as if the record were empty.
-//
-// In general, this is a very tolerant of problems  reader.
-func (reader *Input) ReadRecord() []string {
-	var row []string
-	var fileErr error
-
-	if reader.firstRow != nil {
-		row = reader.firstRow
-		reader.firstRow = nil
-		return row
-	}
+	// What the outputted JSON will be delimited by .
+	RecordDelimiter string
 
-	row, fileErr = reader.reader.Read()
-	emptysToAppend := reader.minOutputLength - len(row)
-	if fileErr == io.EOF || fileErr == io.ErrClosedPipe {
-		return nil
-	} else if _, ok := fileErr.(*csv.ParseError); ok {
-		emptysToAppend = reader.minOutputLength
-	}
+	// Size of incoming object
+	StreamSize int64
 
-	if emptysToAppend > 0 {
-		for counter := 0; counter < emptysToAppend; counter++ {
-			row = append(row, "")
-		}
-	}
+	//To be implemented
```

Comment:
name `Type` is not very intuitive, maybe this can be removed and added later, when you have an implementation ready.

Reply:
Done